compiler/luci/pass/src/SubstituteTransposeToReshapePass.test.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "luci/Pass/SubstituteTransposeToReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+#include <gtest/gtest.h>
+
+namespace
+{
+
+void create_substitute_transpose_to_reshape(loco::Graph *g,
+                                            const std::initializer_list<uint32_t> shape,
+                                            const std::vector<int32_t> perm)
+{
+  assert(g);
+  // Input Create.
+  auto input = g->nodes()->create<luci::CircleInput>();
+  auto graph_input = g->inputs()->create();
+  input->index(graph_input->index());
+  input->shape_status(luci::ShapeStatus::VALID);
+  input->rank(shape.size());
+  input->shape(shape);
+
+  // Permutation Create.
+  auto perm_const = g->nodes()->create<luci::CircleConst>();
+  perm_const->dtype(loco::DataType::S32);
+  perm_const->size<loco::DataType::S32>(perm.size());
+  perm_const->shape_status(luci::ShapeStatus::VALID);
+  perm_const->rank(1);
+  perm_const->dim(0).set(perm.size());
+  for (uint32_t i = 0; i < static_cast<uint32_t>(perm.size()); i++)
+  {
+    perm_const->at<loco::DataType::S32>(i) = perm.at(i);
+  }
+
+  // Transpose Create.
+  auto transpose_node = g->nodes()->create<luci::CircleTranspose>();
+  transpose_node->a(input);
+  transpose_node->perm(perm_const);
+
+  // Output Connect.
+  auto output = g->nodes()->create<luci::CircleOutput>();
+  output->from(transpose_node);
+  auto graph_output = g->outputs()->create();
+  output->index(graph_output->index());
+}
+
+} // namespace
+
+TEST(SubstituteTransposeToReshapePass, simple_case)
```

Comment:
Can you use `TEST_F` macro?  You can refer #5713 .

Reply:
Okay, I will update after refer that.

Reply:
FYI, this is not constant folding, so `ConstantFoldingTestGraph` used in #5713 may not be directly applicable. There are some useful tutorials about gtest fixture written in Korean. (e.g., https://jakejang.tistory.com/39)